Source/Scene/Scene.js

```diff
@@ -2731,7 +2732,16 @@ define([
                 pickedPrimitives.push(primitive);
             }
 
+            prevPickedResult = pickedResult;
             pickedResult = this.pick(windowPosition);
+
+            //>>includeStart('debug', pragmas.debug);
+            if (pickedResult === prevPickedResult) {
+                throw new DeveloperError(
```

Comment:
I excluded commit with adding `DeveloperError` from the pull request.

Are you sure there is no benefit in adding such internal consistency check? (I can open second PR with this change.)
IMO this check costs almost nothing, but can improve debugging experience next time when pick issue will arise.